src/test/xml_results.go

```diff
@@ -434,9 +436,26 @@ func SerialiseResultsToXML(target *core.BuildTarget, indent bool) []byte {
 }
 
 // uploadResults uploads test results to a remote server.
-func uploadResults(target *core.BuildTarget, url string) error {
+func uploadResults(target *core.BuildTarget, url string, gzipped bool) error {
 	b := SerialiseResultsToXML(target, true)
-	resp, err := http.Post(url, "application/xml", bytes.NewReader(b))
+	var r io.Reader = bytes.NewReader(b)
+	if gzipped {
+		var buf bytes.Buffer
+		zw := gzip.NewWriter(&buf)
+		if _, err := io.Copy(zw, r); err != nil {
+			return fmt.Errorf("Failed to gzip test results: %s", err)
+		} else if err = zw.Close(); err != nil {
+			return fmt.Errorf("Failed to flush gzip writer: %s", err)
+		}
+		r = &buf
+	}
+	req, err := http.NewRequest(http.MethodPost, url, r)
+	if err != nil {
+		return fmt.Errorf("Failed to create HTTP request: %s", err)
+	}
+	req.Header["Content-Type"] = []string{"application/xml"}
+	req.Header["Content-Encoding"] = []string{"gzip"}
```

Comment:
should this be in the if gzipped section?

Reply:
🤦 Yes it should, my bad.